CONTRIBUTING.md

```diff
@@ -1,49 +1,179 @@
 
 ## Architecture Decision Records (ADR)
 
-When proposing an architecture decision for ICS, please start by opening an [issue](https://github.com/cosmos/interchain-security/issues/new/choose) with a summary of the proposal. Once the proposal has been discussed and there is rough alignment on a high-level approach to the design, you may either start development, or write an ADR.
+When proposing an architecture decision for ICS, please start by opening an [issue](https://github.com/cosmos/interchain-security/issues/new/choose) or a [discussion](https://github.com/cosmos/interchain-security/discussions/new) with a summary of the proposal. Once the proposal has been discussed and there is rough alignment on a high-level approach to the design, you may either start development, or write an ADR.
 
 If your architecture decision is a simple change, you may contribute directly without writing an ADR. However, if you are proposing a significant change, please include a corresponding ADR.
 
 In certain circumstances, the architecture decision may require changes to the ICS spec. Note that the spec is responsible for defining language-agnostic, implementation-agnostic behaviors for the ICS protocol. Whereas ADRs are responsible for communicating implementation decisions contained within this repo.
 
-To create an ADR, follow the [template](https://github.com/cosmos/interchain-security/blob/main/docs/architecture/adr-template.md) and [doc](https://github.com/cosmos/interchain-security/blob/main/docs/architecture/README.md). If you would like to see examples of how these are written, please refer to the current [ADRs](https://github.com/cosmos/interchain-security/tree/main/docs/architecture).
+To create an ADR, follow the [template](./docs/docs/adrs/adr-template.md) and [doc](./docs/docs/adrs/Readme.md). If you would like to see examples of how these are written, please refer to the current [ADRs](https://github.com/cosmos/interchain-security/tree/main/docs/architecture).
 
 ### ADR Proposals
 
 Architecture Decision Records (ADRs) may be proposed by any contributors or maintainers of ICS. ADRs are intended to be iterative, and may be merged into `main` while still in a `Proposed` status
 
-## PR Targeting
+## Development Procedure
+
+* The latest state of development is on `main`.
+* `main` must never fail `make lint test`.
+* No `--force` onto `main` (except when reverting a broken commit, which should seldom happen).
+* Create a branch to start work:
+    * Fork the repo (core developers must create a branch directly in the ICS repo),
+    branch from the HEAD of `main`, make some commits, and submit a PR to `main`.
+    * For core developers working within the `interchain-security` repo, follow branch name conventions to ensure clear
+    ownership of branches: `{moniker}/{issue#}-branch-name`.
+    * See [Branching Model](#branching-model-and-release) for more details.
+* Be sure to run `make format` before every commit. The easiest way
```

Comment:
@smarshall-spitzbart please add a `format` target to the Makefile

Reply:
See https://github.com/cosmos/interchain-security/pull/860, it's similar to gaia but not completely the same